src/animation/gravity.ts

```diff
@@ -123,3 +114,64 @@ export const gravity = (params: GravityParams): [() => void] => {
 
   return [stop];
 };
+
+interface Gravity2DParams {
+  config: {
+    attractorMass: number;
+    moverMass: number;
+    attractorPosition: Vector<number>;
+    initialMoverPosition?: Vector<number>;
+    initialMoverVelocity?: Vector<number>;
+    threshold?: {
+      min: number;
+      max: number;
+    };
+  };
+  onUpdate: (values: {
+    position: Vector<number>;
+    velocity: Vector<number>;
+  }) => void;
+}
+
+export const gravity2D = (params: Gravity2DParams): [() => void] => {
+  const state: GravityState = {
+    mover: {
+      mass: params.config.moverMass,
+      acceleration: [0, 0],
+      velocity: params.config.initialMoverVelocity || [0, 0],
+      position: params.config.initialMoverPosition || [0, 0]
+    },
+    attractor: {
+      mass: params.config.attractorMass,
+      acceleration: [0, 0],
+      velocity: [0, 0],
+      position: params.config.attractorPosition
+    }
+  };
+
+  const { stop } = rAF().start((timestamp, lastFrame) => {
+    /**
+     * Determine the number of milliseconds elapsed between the current frame
+     * and the last frame. If more than four frames have been dropped, assuming
+     * a 60fps interval, just use the timestamp of the current frame.
+     */
+    const diffTime = timestamp > lastFrame + 64 ? 0 : lastFrame;
+    const steps = Math.floor(timestamp - diffTime);
+
+    if (steps > 0) {
+      // Apply the gravitational force once for each step.
+      for (let i = 0; i < steps; i++) {
+        state.mover = applyForceForStep(state, params.config.threshold);
+      }
+    } else {
+      state.mover = applyForceForStep(state, params.config.threshold);
+    }
+
+    params.onUpdate({
```

Comment:
Notice we have no physics condition stopping the animation for `gravity2D`. Right now, I can't think of a logical reason to do this in library land. We hand back the `stop` function to the user so they can stop it according to their own userland logic, but for us to stop it preemptively doesn't strike me right now as useful. The one case I can see is if they supply `translate: transform(x, y)` and they want a specific `from` / `to` value, but I'm going to punt on that for now since I think we can add a flag – something like `infinite` – to help ease this.